Platon/backend/app/search_engine/views.py

```diff
@@ -0,0 +1,149 @@
+from flask import make_response,jsonify,request
+from flask_restplus import Resource,Namespace, fields
+from sqlalchemy import func
+
+from app.search_engine.forms import UserSearchForm, user_search_parser
+from app.search_engine.helpers import SearchEngine, SearchType
+from app.profile_management.models import Jobs,Skills,UserSkills
+from app.auth_system.models import User
+from app.auth_system.helpers import decode_token
+
+from app import api, db
+import math
+
+search_engine_ns = Namespace("Search Engine",
+                            description="Search Engine Endpoints",
+                            path = "/search_engine")
+
+search_user_model = api.model('User Data', {
+    "id": fields.Integer,
+    "name": fields.String,
+    "surname": fields.String,
+    "profile_photo": fields.String,
+    "e_mail": fields.String,
+    "job_id": fields.Integer
+})
+
+search_user_list_model = api.model('User List(Search)', {
+    'number_of_pages': fields.Integer,
+    'result_list': fields.List(
+        fields.Nested(search_user_model)
+    )
+})
+
+@search_engine_ns.route("/user")
+class UserSearchAPI(Resource):
+    
+    @api.doc(responses={401 : 'Account Problems', 400 : 'Input Format Error' ,500 : ' Database Connection Error'})
+    @api.response(200, 'Valid Search Result', search_user_list_model)
+    @api.expect(user_search_parser)
+    def get(self):
+
+        form = UserSearchForm(request.args)
+        if form.validate():
+            search_query = form.search_query.data.lower()
+            # Remove the punctuation in the search query
+            search_query = SearchEngine.remove_punctuation(search_query)
+            # Tokenize the search query
+            tokens = search_query.split()
+            # Get stopwords form the API
+            stopwords = SearchEngine.get_stopwords()
+            # Remove stopwords from the list of tokens
+            tokens = list(set(tokens) - set(stopwords))
+            # Get the semantically related list for our tokens
+            tokens = SearchEngine.semantic_related_list(tokens)
+            result_list = []
+            result_id_score = []
+            # Search tokens in Jobs Table
+            for token,score in tokens:
+                try:
+                    related_job = Jobs.query.filter(func.lower(Jobs.name) == token).first()
+                except: 
+                    return make_response(jsonify({"error": "Database Connection Problem."}), 500)
+                if related_job is not None:
+                    try:
+                        owner_list = User.query.filter(User.job_id == related_job.id).all()
+                    except: 
+                        return make_response(jsonify({"error": "Database Connection Problem."}), 500)
+                    for user in owner_list:
+                        id_list = [i[0] for i in result_id_score]
```

Comment:
Can you add comments about usage of lists named as result_list, result_id_score and id_list? I couldn't follow your logic.

Reply:
One of the lists sttores all the dictionaries that ise used to store user information. I used other list to make sorting (according to the relevance) easier.